src/jpath.go

```diff
@@ -54,19 +54,26 @@ func main() {
 		os.Exit(1)
 	}
 
-	// parse the input
-	jsonb := input.ParseInputJson(json)
+	// parse the input json
+	jsonb, err := input.ParseInputJson(json)
+	if err != nil {
+		common.ExitWithError(common.InvalidJson)
+	}
+	// parse the expression
 	parsedOutput, err := parser.ProcessExpression(expr, jsonb)
 	if err != nil {
 		if strings.Contains(err.Error(), common.InvalidExpr.GetMsg()) {
```

Comment:
Is this check is for exit with specific error code based on error? We can custom error to include the error code and message to avoid string comparison.

Reply:
Correct, this check is to handle different types of errors differently. 
How do I do custom error code and message checks?

Reply:
Merging this issue to move on with dev. I will address the above code comment in a later commit